strategies/affected/src/main/java/org/arquillian/smart/testing/strategies/affected/detector/FileSystemTestClassDetector.java

```diff
@@ -14,54 +14,34 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
+import org.arquillian.smart.testing.filter.TestVerifier;
 
 /**
  * Implementation that gets all tests using glob pattern () by default and using root directory as reference by default
  */
 public class FileSystemTestClassDetector implements TestClassDetector {
 
     private File rootDirectory;
-    private List<String> globPatterns;
+    private TestVerifier verifier;
 
-    public FileSystemTestClassDetector(File rootDirectory, String... globPatterns) {
+    public FileSystemTestClassDetector(File rootDirectory, TestVerifier verifier) {
         this.rootDirectory = rootDirectory;
-        if (globPatterns.length > 0) {
-            this.globPatterns = Arrays.asList(globPatterns);
-        } else {
-            this.globPatterns = Collections.singletonList("**/src/test/java/**/*Test*.*");
-        }
+        this.verifier = verifier;
     }
 
     @Override
     public Set<File> detect() {
-
-        final Set<File> tests = new HashSet<>();
-
-        for (String pattern : globPatterns) {
-            try {
-                tests.addAll(findTests(rootDirectory, getPathMatcher(pattern)));
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
-        }
-
-        return tests;
-    }
-
-    private Collection<? extends File> findTests(File rootDirectory, PathMatcher pathMatcher) throws IOException {
         try (Stream<Path> stream = Files.walk(rootDirectory.toPath())) {
             return stream
-                .filter(pathMatcher::matches)
+                .filter(path -> !Files.isDirectory(path) && isJavaFile(path) && verifier.isTest(path))
```

Comment:
Why not using `Files.isRegularFile` instead of the negation of the directory check